src/bp/core/services/ghost/service.ts

```diff
@@ -77,23 +80,26 @@ export class GhostService {
     }
 
     const allChanges = await this.listFileChanges(tmpFolder)
-    for (const { scope, changes } of allChanges) {
-      const dbRevs = await this.dbDriver.listRevisions(scope === 'global' ? 'data/global' : 'data/bots/' + scope)
+    for (const { botId, changes, localFiles } of allChanges) {
+      const dbRevs = await this.dbDriver.listRevisions(botId ? 'data/bots/' + botId : 'data/global')
       await Promise.each(dbRevs, rev => this.dbDriver.deleteRevision(rev.path, rev.revision))
 
       await Promise.map(changes.filter(x => x.action === 'del'), async file => {
         await this.dbDriver.deleteFile(file.path)
         await invalidateFile(file.path)
       })
 
-      await Promise.map(changes.filter(x => ['add', 'edit'].includes(x.action)), async file => {
-        const content = await this.diskDriver.readFile(path.join(tmpFolder, file.path))
-        await this.dbDriver.upsertFile(file.path, content, false)
-        await invalidateFile(file.path)
-      })
+      // Upload all local files for that scope
+      if (localFiles.length) {
+        await Promise.map(localFiles, async filePath => {
+          const content = await this.diskDriver.readFile(path.join(tmpFolder, filePath))
+          await this.dbDriver.upsertFile(filePath, content, false)
+          await invalidateFile(filePath)
+        })
+      }
     }
 
-    return allChanges.map(c => c.scope).filter(c => c !== 'global')
+    return allChanges.filter(x => x.localFiles.length && x.botId).map(x => x.botId)
```

Comment:
you removed the `!== 'global'` I assume you handle this somewhere else ?

Reply:
I changed that part of logic, because it was causing trouble if you had a bot called 'global'. So, an undefined bot id is global now